x-pack/plugins/observability/public/components/shared/exploratory_view/configurations/lens_attributes.ts

```diff
@@ -646,8 +649,18 @@ export class LensAttributes {
 
   getMainYAxis(layerConfig: LayerConfig, layerId: string, columnFilter: string) {
     const { breakdown } = layerConfig;
-    const { sourceField, operationType, label, timeScale } =
-      layerConfig.seriesConfig.yAxisColumns[0];
+    const {
+      sourceField,
+      operationType,
+      label: colLabel,
+      timeScale,
+    } = layerConfig.seriesConfig.yAxisColumns[0];
+
+    let label = layerConfig.name || colLabel;
```

Comment:
I think this could be cleaner if we use the ternary operator instead of `let` and reassigning in the conditional block. But it's really a matter of opinion.